packages/core/upload/admin/src/pages/App/MediaLibrary.js

```diff
@@ -105,19 +108,32 @@ export const MediaLibrary = () => {
 
   const assets = assetsData?.results;
   const assetCount = assets?.length ?? 0;
-
   const isNestedFolder = !!query?.folder;
+
   const isLoading =
     foldersLoading || folderStructureIsLoading || permissionsLoading || assetsLoading;
+  const isFolder = !isLoading && query.folder;
+
+  const parentFolderId = isFolder && findRecursiveParentFolderId(folderStructure[0], query.folder);
+  const backQuery = parentFolderId
+    ? stringify({ ...query, folder: parentFolderId }, { encode: false })
+    : stringify(
+        { sort: query.sort, page: query.page, pageSize: query.pageSize },
+        { encode: false }
+      );
+
+  const currentFolder =
+    isFolder && findRecursiveFolderByValue(folderStructure, parseInt(query.folder, 10));
+  const folderLabel = currentFolder && currentFolder.label;
 
   return (
     <Layout>
       <Main aria-busy={isLoading}>
         <HeaderLayout
-          title={formatMessage({
+          title={`${formatMessage({
             id: getTrad('plugin.name'),
-            defaultMessage: 'Media Library',
-          })}
+            defaultMessage: `Media Library`,
+          })}${folderLabel ? ` - ${folderLabel}` : ''}`}
```

Comment:
Do you think we should truncate the `folderLabel` at a certain length? Or does the `HeaderLayout` component already take care of that?

Reply:
I think we need to truncate it, good catch 😅

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/71838159/169334769-858cb191-6295-4f69-9b8b-eac15f9c80ec.png">